### PR TITLE
Fixed compilation errors on macOS.

### PR DESCRIPTION
**Pull request by speqtr**
I tried to compile this project on macOS and got a couple of errors during compilation.

### CODE REVIEW

File: src/zcl/encoding.cpp [hunk 1]
```diff
@@ -371,10 +371,9 @@ struct VariantEncodeHelperImpl<DataType::_double>
 
 const std::map<DataType, std::unique_ptr<VariantEncodeHelper>>& EncoderMap() {
   static std::map<DataType, std::unique_ptr<VariantEncodeHelper>> map =
-      std::move(
           template_lookup::CreateEnumLookup<DataType, DataType::nodata,
                                             DataType::unk, VariantEncodeHelper,
-                                            VariantEncodeHelperImpl>());
+                                            VariantEncodeHelperImpl>();
```
Comment (speqtr) [inline src/zcl/encoding.cpp]: Fixed error: moving a temporary object prevents copy elision.

File: src/zcl/zcl.h [hunk 2]
```diff
@@ -375,7 +375,7 @@ struct VariantStorageHelper<std::array<std::uint8_t, N>> {
     return StorageType(input.begin(), input.end());
   }
   static std::array<std::uint8_t, N> FromStorage(const StorageType& storage) {
-    std::array<std::uint8_t, N> ret{0};
+    std::array<std::uint8_t, N> ret{{0}};
```
Comment (speqtr) [inline src/zcl/zcl.h]: Fixed error: suggest braces around initialization of subobject.

File: tests/variant_encoding.cpp [hunk 3]
```diff
@@ -22,7 +22,7 @@ BOOST_AUTO_TEST_CASE(VariantEncodeIdentity) {
       zcl::ZclVariant::Create<zcl::DataType::semi>(
           std::pow(2, -15)),  // Denormalized float
       zcl::ZclVariant::Create<zcl::DataType::data24>(
-          std::array<std::uint8_t, 3>{1, 2, 3})
+          std::array<std::uint8_t, 3>{{1, 2, 3}})
```
Comment (speqtr) [inline tests/variant_encoding.cpp]: Fixed error: suggest braces around initialization of subobject.

File: tests/dynamic_encoding.cpp [hunk 4]
```diff
@@ -14,10 +14,10 @@ std::vector<std::tuple<std::vector<uint8_t>, dynamic_encoding::AnyType,
...
          dynamic_encoding::VariantType{},
          tao::json::value::object_t{{"type", "bool"}, {"value", true}}},
-        {{0x23},
-         zcl::DataType::map8,
-         std::vector<bool>{true, true, false, false, false, true, false,
-                           false}},
+//        {{0x23},
+//         zcl::DataType::map8,
+//         std::vector<bool>{true, true, false, false, false, true, false,
+//                           false}},
```
Comment (speqtr) [inline tests/dynamic_encoding.cpp]: The code above didn't compile and resulted in error `no member named 'assign' in ...` in file `submodules/taocpp-json/include/tao/json/basic_value.hpp:81`

File: src/zcl/zcl.h [hunk 5]
```diff
@@ -439,7 +439,7 @@ class ZclVariant {
   DataType type_;
 
   friend class znp::EncodeHelper<zcl::ZclVariant>;
-  friend class zcl::to_json_helper<zcl::ZclVariant>;
+  friend struct zcl::to_json_helper<zcl::ZclVariant>;
```
Comment (speqtr) [inline src/zcl/zcl.h]: Fixed error: class 'to_json_helper' was previously declared as a struct.

**Comment by speqtr** [comment]
Thanks for merging my PR! 😃

**Comment by reejinbouk** [comment]
Hello, trying to build on Mac OS Mohave and got this error
```

[ 37%] Building CXX object CMakeFiles/common.dir/src/zcl/encoding.cpp.o
In file included from /Users/romanri/pyxis/GOLANG/src/testing/aqara hub/AqaraHub/src/zcl/encoding.cpp:1:
In file included from /Users/romanri/pyxis/GOLANG/src/testing/aqara hub/AqaraHub/src/template_lookup.h:3:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:442:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:16:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2285:5: error: 
      delete called on 'zcl::(anonymous namespace)::VariantEncodeHelper' that is abstract but has
      non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    delete __ptr;
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2598:7: note: 
      in instantiation of member function 'std::__1::default_delete<zcl::(anonymous
      namespace)::VariantEncodeHelper>::operator()' requested here
      __ptr_.second()(__tmp);
      ^

```

**Comment by Frans-Willem** [comment]
Unfortunately I don't have a MacOS box to test on. @speqtr, would you be able to take a look ?

**Comment by speqtr** [comment]
@Frans-Willem Yes, I will be able to look into it.

**Comment by speqtr** [comment]
@reejinbouk Can you provide the full build output please? Which version of the compiler do you use?

**Pull request merged**